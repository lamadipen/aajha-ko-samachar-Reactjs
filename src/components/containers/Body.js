import React from 'react';

const Body = (props) => {
    return (
        <div className="wrapper row-offcanvas row-offcanvas-left">
            <div className="col-xs-1"/>
            <aside className="col-xs-10">
                <section className="content">
                    <div className="row">
                        <section className="col-lg-12 connectedSortable">
                            {props.children}
                        </section>
                    </div>
                </section>
            </aside>
            <div className="col-xs-1"/>
        </div>
    )
}

export default Body;