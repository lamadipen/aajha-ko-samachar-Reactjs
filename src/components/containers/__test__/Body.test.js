import React from 'react';
import {shallow} from 'enzyme';
import Body from '../Body';

describe('BodyTest', () => {
    let wrapper;

    beforeEach(() => {
        console.log('Before Each ');
        wrapper = shallow(<Body/>);
    });

    it('render without crahsing', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('renders children when passed in', () => {
        const wrapper = shallow((<Body>
        <div className="unique"></div>
        </Body>));
        expect(wrapper.contains(<div className="unique"></div>)).toEqual(true);
    })

})