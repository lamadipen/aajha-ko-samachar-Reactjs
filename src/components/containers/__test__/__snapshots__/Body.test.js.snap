// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BodyTest render without crahsing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Body />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col-xs-1"
        />,
        <aside
          className="col-xs-10"
        >
          <section
            className="content"
          >
            <div
              className="row"
            >
              <section
                className="col-lg-12 connectedSortable"
              />
            </div>
          </section>
        </aside>,
        <div
          className="col-xs-1"
        />,
      ],
      "className": "wrapper row-offcanvas row-offcanvas-left",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "col-xs-1",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <section
            className="content"
          >
            <div
              className="row"
            >
              <section
                className="col-lg-12 connectedSortable"
              />
            </div>
          </section>,
          "className": "col-xs-10",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row"
            >
              <section
                className="col-lg-12 connectedSortable"
              />
            </div>,
            "className": "content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <section
                className="col-lg-12 connectedSortable"
              />,
              "className": "row",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "col-lg-12 connectedSortable",
              },
              "ref": null,
              "rendered": null,
              "type": "section",
            },
            "type": "div",
          },
          "type": "section",
        },
        "type": "aside",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "col-xs-1",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-xs-1"
          />,
          <aside
            className="col-xs-10"
          >
            <section
              className="content"
            >
              <div
                className="row"
              >
                <section
                  className="col-lg-12 connectedSortable"
                />
              </div>
            </section>
          </aside>,
          <div
            className="col-xs-1"
          />,
        ],
        "className": "wrapper row-offcanvas row-offcanvas-left",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "col-xs-1",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <section
              className="content"
            >
              <div
                className="row"
              >
                <section
                  className="col-lg-12 connectedSortable"
                />
              </div>
            </section>,
            "className": "col-xs-10",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="row"
              >
                <section
                  className="col-lg-12 connectedSortable"
                />
              </div>,
              "className": "content",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <section
                  className="col-lg-12 connectedSortable"
                />,
                "className": "row",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "col-lg-12 connectedSortable",
                },
                "ref": null,
                "rendered": null,
                "type": "section",
              },
              "type": "div",
            },
            "type": "section",
          },
          "type": "aside",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "col-xs-1",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
